src/Microsoft.Data/StringColumn.cs

```diff
@@ -85,10 +91,18 @@ protected override object GetValue(long startIndex, int length)
 
         protected override void SetValue(long rowIndex, object value)
         {
-            if (value is string)
+            if (value == null || value is string)
             {
                 int bufferIndex = GetBufferIndexContainingRowIndex(ref rowIndex);
+                var oldValue = this[rowIndex];
                 _stringBuffers[bufferIndex][(int)rowIndex] = (string)value;
+                if (oldValue != (string)value)
+                {
+                    if (value == null)
+                        _nullCount++;
+                    if (oldValue == null && _nullCount > 0)
```

Comment:
Can the `oldValue` be `null` AND `_nullCount <= 0`? Maybe that should be an `Assert`?